recipes/halide/build.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# On OSX, we need to ensure we're using conda's gcc/g++
+if [[ `uname` == Darwin ]]; then
+     export CC=gcc
+     export LD=gcc
+     export CXX=g++
```

Comment:
I am working my way forward slowly here, trying to emulate things I saw on https://github.com/conda-forge/staged-recipes/pull/220, but the truth is that I am not so sure how to go about it.